cms/djangoapps/contentstore/management/commands/set_advanced_fields_all_courses.py

```diff
@@ -0,0 +1,248 @@
+
+        # All Courses
+        # Set the following advanced settings:
+        # `catalog_visilibilty` to `about`
+        python manage.py cms --settings=devstack_docker set_advanced_fields_all_courses \
+        --fields_json '{}' --user edx@example.com --course course-v1:ORG+COURSENUMBER+COURSERUN --force-update-catalog-visibility-about # pylint: disable=line-too-long
+    """
+    help = 'Set an advanced settings for all courses based on a json list of fields.'
+
+    fields_json_help = '\
+        Need to have a json list of "field_name": "<new value>" \
+        { \
+            "catalog_visibility": { \
+                "value": "about" \
+            }, \
+            "certificates_display_behavior": { \
+                "value": "early_no_info" \
+            }, \
+            "invitation_only": { \
+                "value": "true" \
+            } \
+        }'
+    user_help = '--user <email> required, needs to be a staff member'
+    course_help = '--course <id> required, e.g. course-v1:ORG+COURSENUMBER+COURSERUN'
+    force_update_catalog_visibility_about_help = '--force-update-catalog-visibility-about, To fix `There was an error loading this course` error when loading MFE learning this forces `catalog_visibility` from `none` to `about`.'  # pylint: disable=line-too-long
+    limit_bulk_update_operation_help = '--limit-bulk-update-operation {0-N}'
+
+    _force_update_catalog_visibility_about = False
+    _updated_catalog_visilibity_about = []
+
+    def add_arguments(self, parser):
+        parser.add_argument('--fields_json',
+                            dest='fields_json',
+                            default=False,
+                            required=True,
+                            help=self.fields_json_help)
+        parser.add_argument('--user',
+                            dest='user_email',
+                            default=False,
+                            required=True,
+                            help=self.user_help)
+        parser.add_argument('--course',
+                            dest='course_id',
+                            default=False,
+                            required=False,
+                            help=self.course_help)
+        parser.add_argument('--force-update-catalog-visibility-about',
+                            dest='force_update_catalog_visibility_about',
+                            action='store_true',
+                            help=self.force_update_catalog_visibility_about_help)
+
+        # WARNING
+        # Limiting the all courses bulk advanced setting update here to avoid consuming all
+        # server memory. Defaulting bulk operations to 150 courses but you can use this argument
+        # to specify any value different than the default.
```

Comment:
Do you know _why_ this consumes so much memory if the # of courses are not limited? From looking at the code, it looks like it loads each course one at a time, updates its settings, then moves on - so (assuming the garbage collector is working as normal), I would expect that only one course is ever loaded in memory at a time, and there would be no limit to how many courses this can process.

Related: if a server has 10,000 courses but can only process 150 at a time, how can they use this tool to do it? I don't see an option to "skip N courses" so that this could be used in batches.